### ISSUE TITLE
[logging] Turn off loss logging, while keeping progress bar and logging to third-party application

### ISSUE THREAD

**Issue by antonioloison** [huggingface/transformers#18093]
### Feature request

I would like to add a training argument to the `TrainingArguments` class to turn off the loss logging to stdout while keeping the progress bar and logging to a third-party application like Weights and Biases.

### Motivation

I am working on a project that trains a model with the Trainer class. I need to log the losses at every epoch to Weights and Biases. Here is my code:
```
training_arguments = TrainingArguments(
    output_dir="./logging_dir",
    num_train_epochs=epochs,
    per_device_train_batch_size=batch_size,
    per_device_eval_batch_size=batch_size,
    report_to="wandb",
    logging_strategy="epoch",
)

trainer = Trainer(
    model=model,
    args=training_arguments,
    tokenizer=tokenizer,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
)
```
This code is contained in a python file `train.py` and is launched in the terminal via the command `python train.py`.
However, I don’t want to have the loss printed because it breaks my progress bar as you can see in the image below and I need the progress bar to give some feedback on the training time.

<img width="1155" alt="cut_progress_bar" src="https://user-images.githubusercontent.com/48316195/178239258-ad61d97f-d851-4c31-a4da-c653aa17db3b.png">

### Your contribution

I think that I could add a `disable_on_log` argument to `TrainingArguments`. Then in the `on_log` of the `ProgressCallback`, a condition should be added like this:

```
def on_log(self, args, state, control, logs=None, **kwargs):
    if state.is_local_process_zero and self.training_bar is not None and args.disable_on_log:
         _ = logs.pop("total_flos", None)
         self.training_bar.write(str(logs))
```

**Comment by LysandreJik** [comment]
You should be able to disable logging while keeping progress bars active. Have you tried setting the log level manually?

```python
from transformers.utils.logging import set_verbosity_error

set_verbosity_error()
```

**Comment by antonioloison** [comment]
Yes, I've tried that, but this makes my progress bar disappear and keeps the loss logging

**Comment by antonioloison** [comment]
Thank you for your answer @LysandreJik. I also tried that, but I want to keep the logs in my third-party application (Weights and Biases here). If I use `logging_strategy="no"`, no more logs are reported to Weights and Biases.

**Comment by DableUTeeF** [comment]
Not sure if it ever solved in the recent releases but for people who stuck in the older versions. As it is [this](https://github.com/huggingface/transformers/blob/main/src/transformers/trainer_callback.py#L516C20-L516C20) line that prints the dict log. While progress bar is handled somewhere else. This is an ugly way to stop `ProgressCallback` from printing the dict without modifying the source (at least in older version).

```
from transformers.trainer_callback import ProgressCallback
def on_log(self, args, state, control, logs=None, **kwargs):
    if state.is_local_process_zero and self.training_bar is not None:
        _ = logs.pop("total_flos", None)
ProgressCallback.on_log = on_log
```

**Comment by pprobst** [comment]
I know this issue is old, but as far as I know, there's no way to achieve what OP intends unless he uses the solution posted above by @DableUTeeF. If there is, I need help finding it in the documentation.

I believe this is important for people that have their own logging solution. Would be cool if there was a clean way to keep logging the loss every `logging_steps`, but disable printing itself.

**Comment by ludekcizinsky** [comment]
I have the exact same problem, and would highly rate such feature, it is super annoying to have my screen spammed with all the logs given that i have all the info report to wandb...

**Comment by FGhavamian** [comment]
This removes the print outs of dictionaries. 
```
from transformers import Trainer, PrinterCallback
trainer = Trainer(...
trainer.remove_callback(PrinterCallback)
```

**Comment by kuromancer** [comment]
Despite calling `grpo_trainer.remove_callback(PrinterCallback)` it's printing this massive completions table and the dict at every step.
How do I make the trainer shut up? I want neither of those while still logging to wandb.
Any help, please?

<img width="742" height="309" alt="Image" src="https://github.com/user-attachments/assets/56fd1d88-89bb-47d3-bf59-efd1dd9dbffd" />

**Comment by simplexsigil** [comment]
Time has passed, but for people coming here:

This is from `Trainer.__init__()`:
```python
self.add_callback(PrinterCallback if self.args.disable_tqdm else DEFAULT_PROGRESS_CALLBACK)
```

so in the last case you might need to remove `ProgressCallback` or better just remove both:
```python
        # Remove default Trainer callbacks that produce unwanted output
        from transformers import PrinterCallback, ProgressCallback
        trainer.remove_callback(PrinterCallback)
        trainer.remove_callback(ProgressCallback)
        print("Removed default Trainer callbacks (PrinterCallback, ProgressCallback)")
 ```
